webapp-django/crashstats/home/static/home/js/home.js

```diff
@@ -0,0 +1,386 @@
+    // of the beta versions starting with that same version number.
+    var betaVersions = [];
+    for (var i = versions.length - 1; i >= 0; i--) {
+        if (versions[i].endsWith('b')) {
+            betaVersions.push(versions[i]);
+        }
+    }
+
+    // Set default dates.
+    var endDate = new Date();
+    endDate.setDate(endDate.getDate() - 1);  // We start with yesterday.
+    var startDate = new Date();
+    startDate.setDate(endDate.getDate() - duration);
+
+    var endDateStr = dateIsoFormat(endDate);
+    var startDateStr = dateIsoFormat(startDate);
+
+    /**
+     * Data received from services, after being transformed, will be stored
+     * here. Once we have received everything, we'll be able to combine all
+     * those data into the final data we want to show as a graph.
+     */
+    var dataStore = {
+        superSearch: null,
+        adi: null,
+        builds: null,
+    };
+
+    /**
+     * Return a string with only the date part of a Date object, in ISO format.
+     */
+    function dateIsoFormat(date) {
+        return date.toISOString().substring(0, 10);
+    }
+
+    /**
+     * History change handler. Reloads the graph's data.
+     */
+    window.onpopstate = function (e) {
+        var days = e.state.duration;
+
+        $('#duration .selected').removeClass('selected');
+        $('#duration .days-' + days).addClass('selected');
+
+        updateDates(days);
+        loadData();
+    };
+
+    /**
+     * Click event handler for the duration menu. Changes the history and
+     * reloads the graph's data.
+     */
+    $('#duration a').click(function (e) {
+        e.preventDefault();
+
+        var link = $(this);
+
+        // Do nothing if this is already selected.
+        if (link.hasClass('selected')) {
+            return;
+        }
+
+        var days = link.data('duration');
+        var url = link.attr('href');
+
+        $('#duration .selected').removeClass('selected');
+        link.addClass('selected');
+
+        window.history.pushState({ duration: days }, pageTitle, url);
+
+        updateDates(days);
+        loadData();
+    });
+
+    /**
+     * Change the `startDate` depending on a duration.
+     */
+    function updateDates(days) {
+        startDate.setDate(endDate.getDate() - days);
+        startDateStr = dateIsoFormat(startDate);
+    }
+
+    /**
+     * Get count of crashes per day and per version.
+     */
+    function querySuperSearch() {
+        var params = {
+            product: product,
+            version: versions,
+            date: [
+                '>=' + startDateStr,
+                '<' + endDateStr,
+            ],
+            '_histogram.date': 'version',
+            _results_number: 0,
+        };
+
+        var url = superSearchApiUrl + '?' + $.param(params, true);
+        $.ajax({
+            url: url,
+            success: onSuperSearchData,
+            error: onQueryError,
+        });
```

Comment:
Cool!

I'm almost wondering if we should hardcode the value for `superSearchApiUrl`. Since it's a fully exposed thing. A URL we're committed to not changing.